build: update several dependencies

alan-agius4 opened the pull request:
**test: remove `cli-hello-world-lazy-rollup` tests**
    
In version 12 of the Angular CLI the `experimentalRollupPass` has been removed.


**build: update Angular CLI packages to 12.0.0-next.7**

With this change we update Angular CLI packages that are used in the repo for testing to `12.0.0-next.7`


**test(ngcc): provide correct source-mappings for renderer tests**

The recent update to MagicString, results in a different basic set of mappings in the renderer. This change updates our tests to match.

**build(docs-infra): update dgeni-packages dependency**

This update includes fixes that prevented us from updating
other dependencies.

**ci: improve renovate configuration**
    
With this change we add several packages to ignored `ignoreDeps` as these packages cannot be updated safely as they cause a large number of errors.
    
In addition, we add a group for `remark` packages.
    
**build: update several dependencies**

With this change we update several dependencies to avoid Renovate from create a lot of PRs during onboarding. We also remove yarn workspaces as after further analysis these are not needed.

Certain dependencies such as `@octokit/rest`, `remark` and `@babel/*` have not been updated as they require a decent amount of work to update, and it's best to leave them for a seperate PR.

alan-agius4 committed:
build: update several dependencies

With this change we update several dependencies to avoid Renovate creating a lot of PRs during onboarding. We also remove yarn workspaces as after further analysis these are not needed.

Certain dependencies such as `@octokit/rest`, `remark` and `@babel/*` have not been updated as they require a decent amount of work to update, and it's best to leave them for a seperate PR.

alan-agius4 committed:
ci: improve renovate configuration

With this change we add several packages to ignored `ignoreDeps` as these packages cannot be updated safely as they cause a large number of errors.

In addition, we add a group for `remark` packages.

alan-agius4 commented on aio/package.json:
```diff
@@ -90,96 +90,101 @@
+  },
+  "// 1": "`dgeni-packages` dependency on `typescript` is too loose which causes an incompatble version to be downloaded.",
+  "// 2": "TODO: remove when it's fixed upstream.",
+  "resolutions": {
```
https://github.com/angular/dgeni-packages/pull/304

alan-agius4 commented on .circleci/config.yml:
```diff
@@ -186,7 +190,11 @@ commands:
+          keys:
+            - *cache_key_win
+            - *cache_key_win_fallback
+      - run: yarn install --frozen-lockfile --non-interactive --cache-folder ../.cache/yarn
```
On windows `~/` doesn't resolve from command line, hence `../` which results in the same path in this case.

petebacondarwin committed:
build(docs-infra): update dgeni-packages dependency

This update includes fixes that prevented us from updating
other dependencies.

petebacondarwin commented:
@googlebot I consent.

petebacondarwin committed:
test(ngcc): provide correct source-mappings for renderer tests

The recent update to MagicString, results in a different basic set of
mappings in the renderer. This change updates our tests to match.

alan-agius4 committed:
build: update Angular CLI packages to 12.0.0-next.7

With this change we update Angular CLI packages that are used in the repo for testing to `12.0.0-next.7`

alan-agius4 commented:
Thanks to our Englishman (@petebacondarwin) for all the help in NGTSC and Dgeni.

alan-agius4 commented:
**Note**: Once this lands you might experience "weird" errors when using `ng-dev` or some tests targets, example

```
node_modules/@octokit/request/node_modules/@octokit/endpoint/dist-node/index.js:24
    if (isPlainObject(options[key])) {
        ^

TypeError: isPlainObject is not a function
```

This is related to incorrect module hoisting, Please run the below:
```
rm -rf node_modules aio/node_modules
yarn
```

Or even better, **but** make sure you don't have any untracked file as you will **loose** them.
```
git clean -dxf
yarn
```

josephperrott commented on packages/zone.js/test/common/zone.spec.ts:
```diff
@@ -334,23 +334,23 @@ describe('Zone', function() {
-        }
-        Zone.assertZonePatched();
-      });
+      // xit('should throw error if ZoneAwarePromise has been overwritten', () => {
```
@JiaLiPassion Should this test be removed or reenabled?

alan-agius4 replied:
Note: The reason why I didn’t spend time fixing the compile time error of this test is that it was disabled.

JiaLiPassion replied:
The case is not correct, I created a PR to fix it. https://github.com/angular/angular/pull/41526

alan-agius4 committed:
test: remove `cli-hello-world-lazy-rollup` tests

In version 12 of the Angular CLI the `experimentalRollupPass` has been removed.

gkalpak commented on package.json:
```diff
@@ -109,72 +99,73 @@
-    "protractor": "^5.4.2",
-    "puppeteer": "5.4.1",
+    "protractor": "^7.0.0",
+    "puppeteer": "8.0.0",
```
The process for updating `puppeteer` is more involved (see [dev-infra/browsers/README.md][1]), so I suggest leaving it out of this PR and also adding it to the list of ignored packages for Renovate.

[1]: https://github.com/angular/angular/blob/cb05fca1ba1d71b326eaad5e38e8ce46a6697fe2/dev-infra/browsers/README.md

alan-agius4 replied:
Agreed, let’s follow on this in a a separate PR.

gkalpak reviewed (changes requested):
Yay! :tada:

gkalpak reviewed (changes requested):
BTW, there are a couple of commit message typos:
- `avoid Renovate from create` --> `avoid Renovate creating`
- `add several package` --> `add several packages`
- `updated safly` --> `updated safely`

alan-agius4 commented on dev-infra/release/publish/test/github-api-testing.ts:
```diff
@@ -61,7 +61,7 @@ export class GithubTestingRepo {
 
   expectChangelogFetch(branch: string, content: string): this {
-    nock(this.repoApiUrl).get(`/contents//CHANGELOG.md`).query(p => p.ref === branch).reply(200, {
+    nock(this.repoApiUrl).get(`/contents/%2FCHANGELOG.md`).query(p => p.ref === branch).reply(200, {
```
Because `nock` will no mock the url correctly otherwise.

Re the double slash, @josephperrott would know more.

josephperrott replied:
The calls to `Octokit`'s `repo.getContents` method all use the path as `/{path}`.  We could update this to just be the relative path.  I am not sure if @devversion had a specific reason for the way it is done now, but both are supported by the API.

devversion replied:
No strong reason, other than trying to be explicit about it being a path from the root of the repository. The change seems reasonable to me. I wonder though what update requires this now. I don't see Octokit/Nock updated.

alan-agius4 replied:
Both of them have been updated indirectly via the lock file generation.

alan-agius4 commented on dev-infra/pr/merge/defaults/integration.spec.ts:
```diff
@@ -55,7 +55,7 @@ describe('default target labels', () => {
   function interceptBranchVersionRequest(branchName: string, version: string) {
     nock(getRepoApiRequestUrl())
-        .get('/contents//package.json')
+        .get('/contents/%2Fpackage.json')
```
Same as above

alan-agius4 commented on renovate.json:
```diff
@@ -57,7 +70,15 @@
           "bazel_toolchains"
         ],
-        "groupName": "Bazel",
+        "groupName": "bazel",
+        "pinVersions": false
+      },
+      {
+        "packagePatterns": [
+          "^remark-.*",
+          "remark"
+        ],
+        "groupName": "remark",
```
Yes.

alan-agius4 replied:
I can create a separate commit or rename the description of this one.

alan-agius4 commented on scripts/puppeteer-chromedriver-versions.js:
```diff
@@ -13,6 +13,7 @@
 // "chromium_revision": "722234" but this does not map easily to the Chrome version
 // so we use this mapping here instead.
 module.exports = {
+  '8.0.0': '89.0.4389.23',
```
Oops

alan-agius4 commented on packages/zone.js/test/browser/browser.spec.ts:
```diff
@@ -2555,17 +2555,19 @@ describe('Zone', function() {
            (Zone as any)[zoneSymbol('ignoreConsoleErrorUncaughtError')] = true;
            Zone.root.fork({name: 'promise'}).run(function() {
-             const listener1 = (evt: any) => {
+             function listener1(evt: any) {
```
Without these changes, this test failed. The default behaviour was not prevented which caused the error to bubble up.

alan-agius4 replied:
@JiaLiPassion can you please take a look?

gkalpak replied:
Just to be clear, I am not asking about the addition of `evt.preventDefault()` but the switch from arrow functions to regular functions.

alan-agius4 replied:
Yeah, I know, both are needed.

JiaLiPassion replied:
I tested it, it seems we don't need to update this case, the reason is after jasmine new version, jasmine defined a `window.onerror` handler which is overwritten by other cases.   I created another PR to fix it. https://github.com/angular/angular/pull/41526

JiaLiPassion replied:
I will fix this case and the related case laster, @alan-agius4 , could you please also comment out this case?

alan-agius4 replied:
I disabled the test.

alan-agius4 commented on goldens/size-tracking/aio-payloads.json:
```diff
@@ -3,26 +3,26 @@
         "runtime-es2015": 3033,
-        "main-es2015": 450953,
-        "polyfills-es2015": 52215
+        "main-es2015": 451829,
+        "polyfills-es2015": 55228
```
Mainly the Angular and Zone.JS package updates.

Julien-Marcou commented:
Even though it is not required, I believe it would be better to update TypeScript to v4.2.4 for these files too, so it's easier to track TypeScript version usage for futur updates :

- aio/aio-builds-setup/dockerbuild/scripts-js/package.json
- aio/aio-builds-setup/dockerbuild/scripts-js/yarn.lock 
- integration/typings_test_ts42/package.json
- packages/bazel/package.json 
- packages/compiler-cli/package.json 
- packages/zone.js/package.json
- packages/zone.js/test/typings/package.json
- packages/zone.js/yarn.lock

And also update `packages/compiler-cli/src/typescript_support.ts` with `const MIN_TS_VERSION = '4.2.4';`

josephperrott reviewed (approved):
LGTM

gkalpak commented on renovate.json:
```diff
@@ -23,16 +23,31 @@
       "tools/ts-api-guardian/package.json",
-      "aio/package.json"
+      "aio/package.json",
+      "aio/tools/examples/shared/package.json"
```
I think it would be better to exclude `aio/tools/examples/shared/package.json`. This file is used for the docs examples and it must (a) remain in-sync with the `package.json` files in [aio/tools/examples/shared/boilerplate/][1] and (b) reflect as close as possible a CLI generated app (for the CLI version currently used for docs examples).

(For reference, the updating process is described in [aio/tools/examples/UPDATING.md][2].)

[1]: https://github.com/angular/angular/blob/10a7c876929683234c6331905301727b1b337674/aio/tools/examples/shared/boilerplate
[2]:
https://github.com/angular/angular/blob/10a7c876929683234c6331905301727b1b337674/aio/tools/examples/UPDATING.md

gkalpak reviewed (approved):
One last comment (about excluding `aio/tools/examples/shared/package.json`). Otherwise lgtm 🚀

(I would still be curious to know the answer to https://github.com/angular/angular/pull/41434#discussion_r609011902, but this should not block the PR.)

alan-agius4 committed:
fixup! ci: improve renovate configuration

alan-agius4 commented:
Looks like this change https://github.com/angular/angular/commit/deacc741e0ee41666292d2e2c07812a78daf5d6f by @alxhub caused a breaking change in the compiler API. Therefore, we need to wait for the next CLI release to unblock this PR.

alan-agius4 committed:
fixup! build: update several dependencies

AndrewKushnir commented on goldens/size-tracking/integration-payloads.json:
```diff
@@ -39,22 +39,12 @@
       "uncompressed": {
         "runtime-es2015": 2285,
-        "main-es2015": 240883,
+        "main-es2015": 241389,
```
@alan-agius4 just want to check whether this increase is caused by the same reasons as described in https://github.com/angular/angular/pull/41434/files#r609355897 (we don't have RxJs version change in the root `package.json` file though)?

alan-agius4 replied:
RxJS did change, because will the version was not updated in the package.json, we are using another version because the lock file was updated.

AndrewKushnir reviewed (approved):
LGTM, just a quick question on payload size change.

JiaLiPassion reviewed (approved):
LGTM, I will fix the commented zone.js test cases after this PR is merged.

petebacondarwin commented on goldens/size-tracking/aio-payloads.json:
```diff
@@ -3,26 +3,26 @@
-        "main-es2015": 437306,
-        "polyfills-es2015": 52493
+        "main-es2015": 438257,
+        "polyfills-es2015": 55228
```
On master this value is `52215` (same as ivy).

petebacondarwin commented on goldens/size-tracking/aio-payloads.json:
```diff
@@ -3,26 +3,26 @@
         "runtime-es2015": 3153,
-        "main-es2015": 437306,
-        "polyfills-es2015": 52493
+        "main-es2015": 438257,
```
```suggestion
        "main-es2015": 438598,
```

On master this value is `437622`.

petebacondarwin commented on goldens/size-tracking/aio-payloads.json:
```diff
@@ -3,26 +3,26 @@
         "runtime-es2015": 3033,
-        "main-es2015": 451600,
-        "polyfills-es2015": 52215
+        "main-es2015": 452581,
```
```suggestion
        "main-es2015": 452892,
```

petebacondarwin reviewed (approved):
Reviewed-for: size-tracking

alan-agius4 committed:
fixup! build: update several dependencies

alan-agius4 committed:
build: update `@angular-devkit/build-angular` to use commit SHA

This is a temporary workaround until the CLI version containing a fix for the regression caused by deacc74 is available on NPM.

Without this change CLI builds will fail with;
```
angularCompiler.getNextProgram is not a function
```

alan-agius4 committed:
fixup! build: update several dependencies

alan-agius4 committed:
fixup! build: update `@angular-devkit/build-angular` to use commit SHA

alan-agius4 committed:
test: update CLI integration tests to run only for Ivy

In Angular CLI 12, application can only be compiled using Ivy, therefore we shouldn't run these tests when Bazel runs with View Engine context.

alan-agius4 committed:
fixup! build: update several dependencies

alan-agius4 committed:
fixup! test: update CLI integration tests to run only for Ivy

AndrewKushnir reviewed (approved):
Reviewed-for: size-tracking
